group-coordinator/src/main/java/org/apache/kafka/coordinator/group/assignor/AssignmentSpec.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.coordinator.group.assignor;
+
+import org.apache.kafka.common.Uuid;
+
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * The assignment specification for a consumer group.
+ */
+public class AssignmentSpec {
+    /**
+     * The members keyed by member id.
+     */
+    final Map<String, AssignmentMemberSpec> members;
+
+    /**
+     * The topics' metadata keyed by topic id
+     */
+    final Map<Uuid, AssignmentTopicMetadata> topics;
+
+    public AssignmentSpec(
+        Map<String, AssignmentMemberSpec> members,
+        Map<Uuid, AssignmentTopicMetadata> topics
+    ) {
+        Objects.requireNonNull(members);
+        Objects.requireNonNull(topics);
+
```

Comment:
extremely small nit: any reason why we include a newline here but not on the other specs

Reply:
Will remove it.